service-base/src/main/java/org/eclipse/hono/service/credentials/validators/CredentialsValidatorHashedPassword.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.service.credentials.validators;
+
+import java.io.UnsupportedEncodingException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+
+import org.eclipse.hono.util.CredentialsConstants;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Validator to validate credentials of type {@link CredentialsConstants#SECRETS_TYPE_HASHED_PASSWORD} for a given password.
+ */
+@Component
+@Scope("prototype")
+public final class CredentialsValidatorHashedPassword extends AbstractCredentialsValidator<String> {
```

Comment:
Scope singleton: +1
static : IMHO this does not make a difference, since there is a singleton instance of it. So I will keep it like it is (in the future there might be reasons to not have all static, so since there is no need for it, I like to avoid static if possible).